buildkite/resource_pipeline.go

```diff
@@ -3,17 +3,19 @@ package buildkite
 import (
 	"context"
 	"fmt"
-	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
-	"github.com/shurcooL/graphql"
 	"log"
 	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/shurcooL/graphql"
 )
 
 // PipelineNode represents a pipeline as returned from the GraphQL API
 type PipelineNode struct {
 	CancelIntermediateBuilds             graphql.Boolean
 	CancelIntermediateBuildsBranchFilter graphql.String
 	DefaultBranch                        graphql.String
+	BranchConfiguration                  graphql.String
```

Comment:
@jradtilbrook despite of having this new field for this struct, I still get the following error when using the plugin:

```
Error: Field 'branchConfiguration' doesn't exist on type 'Pipeline'
```

Wondering if I miss anything here. Many thanks

Reply:
It looks to me like this isn't available in the GraphQL API. You can explore the GraphQL API and run queries from https://graphql.buildkite.com/explorer.

You can see I have gotten around this for deleting a pipeline which at the time of development wasn't available - but I now see it is :tada:! So if you want to use the http client that is available to update the branch configuration when creating or updating a pipeline you could do that in the meantime. See https://github.com/jradtilbrook/terraform-provider-buildkite/blob/master/buildkite/resource_pipeline.go#L252-L259 for how I've done it previously.

Pinging @sj26 to see if they can confirm.

Reply:
You are right. There is no `branch_configuration` under `pipeline` in GraphQL API. 

I am going to convert the code to http api instead.

Reply:
Confirmed, there's no `Pipeline#branchConfiguration` property at the moment, sorry. I'll raise it internally and see if it can be added.

Reply:
No problems, thanks for checking and getting back to us